test/hypothesis_utils.py

```diff
@@ -15,79 +16,113 @@
+_UNDERLYING_TYPE = {
+    torch.quint8: torch.uint8,
+    torch.qint8: torch.int8,
+    torch.qint32: torch.int32
+}
+
+# Enforced zero point for every quantized data type.
+# If None, any zero_point point within the range of the data type is OK.
+_ENFORCED_ZERO_POINT = defaultdict(lambda: None, {
+    torch.quint8: None,
+    torch.qint8: None,
+    torch.qint32: 0
+})
+
+"""Hypothesis filter to avoid overflows with quantized tensors.
 
 Args:
-    shapes: A list of shapes to generate.
-    dtypes: A list of data types to generate. See note below.
-    float_min, float_max: Min and max FP value for the output.
-Generates:
-    Xhy: Tensor of type `float32` and shape drawn from the `shapes`.
-    (scale, zero_point): Drawn from valid ranges derived from the dtypes
-    (qmin, qmax): Valid quantization ranges derived from the dtypes.
-    quantized_type: Data types for conversion in test.
-Note:
-    - The `dtypes` argument is used to infer the ranges. The elements should be
-    of length 1 or 2:
-        If the length is 1 -- The zero_point is not enforced.
-        If the length is 2 -- Argument 0 is torch dtype,
-            the zero_point is also enforced by argument 1
+    tensor: Tensor of floats to filter
+    qparams: Quantization parameters as returned by the `qparams`.
+
+Returns:
+    True
+
+Raises:
+    hypothesis.UnsatisfiedAssumption
+
+Note: This filter is slow. Use it only when filtering of the test cases is
+      absolutely necessary!
 """
-@st.composite
-def qtensor(draw, shapes, dtypes=None, float_min=None, float_max=None):
-    # In case shape is a strategy
-    if isinstance(shapes, SearchStrategy):
-        shape = draw(shapes)
-    else:
-        shape = draw(st.sampled_from(shapes))
-    # Resolve types
-    if dtypes is None:
-        dtypes = ALL_QINT_TYPES
-    _dtypes = draw(st.sampled_from(dtypes))
-    assert len(_dtypes) in [1, 2]
-    if len(_dtypes) == 1:
-        quantized_type = _dtypes[0]
-        _zp_enforce = None
-    elif len(_dtypes) == 2:
-        quantized_type, _zp_enforce = _dtypes[:2]
-    _qtype_info = torch.iinfo(quantized_type)
-    qmin, qmax = _qtype_info.min, _qtype_info.max
-    # Resolve zero_point
-    if _zp_enforce is not None:
-        zero_point = _zp_enforce
-    else:
-        zero_point = draw(st.integers(min_value=qmin, max_value=qmax))
-    if float_min is None or float_max is None:
-        _float_type_info = torch.finfo(torch.float)
-        float_min = _float_type_info.min
-        float_max = _float_type_info.max
-    else:
-        assert float_min <= float_max, 'float_min must be <= float_max'
-    float_eps = _float_type_info.eps
-    # Resolve scale
-    scale = draw(st.floats(min_value=float_eps,
-                           max_value=float_max))
+def assume_not_overflowing(tensor, qparams):
+    (scale, zero_point), (qmin, qmax), quantized_type = qparams
+    adjustment = 1 + torch.finfo(torch.float).eps
+    float_min = tensor.min()
```

Comment:
wait, float_min should be torch.finfo(torch.float).min?

Reply:
Not necessarily. This just checks that the generated tensor passes, not all possible tensors.